src/renderer/contexts/SettingsContext.tsx

```diff
@@ -46,6 +47,7 @@ export const SettingsProvider = memo(({ children }: React.PropsWithChildren<unkn
     const useIsFp16 = useMemoArray(useLocalStorage('is-fp16', false));
     const usePyTorchGPU = useMemoArray(useLocalStorage('pytorch-gpu', 0));
     const useNcnnGPU = useMemoArray(useLocalStorage('ncnn-gpu', 0));
+    const useNcnnBudgetLimit = useMemoArray(useLocalStorage('ncnn-budget-limit', 1024 ** 5));
```

Comment:
Why did you make the default 1 Petabyte?

Reply:
The intent was to pick a default that is practically equivalent to "no limit". I find it very unlikely that any usage will go over 1 PiB; 1 TiB is low enough that some users might hit it (some of my test workflows hit 600 GiB or so, and I'm not as much of a power user as some people).

Reply:
My workstation has around 300 GiB of RAM, and I typically run chaiNNer in a VM on that workstation that has around 200 GiB of RAM assigned to it. I wouldn't be surprised if there exists some power user out there with 1 TB of RAM, and I'm trying to avoid surprising behavior in such (admittedly rare) situations.

That said -- am I correct in assuming that your concern here is that 1 PiB looks "weird" in the GUI as the default, as opposed to having a concern about the default being "no effective additional limit"? If so, would it be OK to instead modify the backend to special-case a limit of 0 so that 0 means "no additional limit", and make 0 the default?